packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -308,4 +341,84 @@ class PreviewCommand extends ShorebirdCommand {
       '${platform.name}_${release.version}.$extension',
     );
   }
+
+  Future<void> setChannelOnRunner({
+    required Directory runnerDirectory,
+    required String channel,
+  }) async {
+    await Isolate.run(() async {
+      final shorebirdYaml = File(
+        p.join(
+          runnerDirectory.path,
+          'Frameworks',
+          'App.framework',
+          'flutter_assets',
+          'shorebird.yaml',
+        ),
+      );
+
+      if (!shorebirdYaml.existsSync()) {
+        throw Exception('Unable to find shorebird.yaml');
+      }
+
+      final yaml = YamlEditor(shorebirdYaml.readAsStringSync())
+        ..update(['channel'], channel);
+
+      shorebirdYaml.writeAsStringSync(yaml.toString(), flush: true);
+    });
+  }
+
+  Future<void> setChannelOnAab({
+    required File aabFile,
+    required String channel,
+  }) async {
+    // Getting the reference here since we cannot inside the isolate.
```

Comment:
Is this because the `scoped` overrides are per-isolate?

Reply:
yup